scripts/javascript.js

```diff
@@ -0,0 +1,163 @@
+  const description =
+    newsitem.description !== null ? `<h3>${newsitem.description}</h3>` : "";
+  const content =
+    newsitem.content !== null
+      ? `<p>${newsitem.content.split("[+")[0]}</p>`
+      : "";
+
+  return `
+    <a href="${url}" class="news--article-link">
+    <figure class="news--article-image" style="background-image: url(${imageurl});" ></figure>
+    ${source}
+    </a>
+    <section class="news--article-content">
+    <header><h2>${title}</h2></header>
+    ${description}
+    ${content}
+    <p><a href="${url}" title="Visit news article: ${title}">Read full article</a></p>
+   <input class="redcard-checkbox" type="checkbox" name="ratenews"> <label>Show article the Red Card!</label>
+    </section>`;
+};
+
+// TOFIX cleared on refresh, due to reset on displayDateOnPage?
+function redCard() {
+  const redcardCheckbox = document.querySelectorAll(".redcard-checkbox");
+
+  return redcardCheckbox.forEach(item => {
+    item.addEventListener("change", function(event) {
+      event.target.checked
+        ? (event.target.parentNode.style.backgroundColor = "red")
+        : (event.target.parentNode.style.backgroundColor = "");
+    });
+  });
+}
+
+function displayDataOnPage(newsStories) {
+  parentNode.innerHTML = "";
+
+  const newsArray = newsStories.articles;
+  // add news blocks (as articles)
+  newsArray.forEach(function(newsitem) {
+    const node = document.createElement("article");
+    node.innerHTML = newsLayout(newsitem);
+    parentNode.appendChild(node);
+  });
+  //
+  redCard();
+}
+
+/*
+------------------
+GENERATE FETCH URL
+------------------
+*/
+
+// news sources object
+let publicationList = {
+  "daily-mail": false,
+  mirror: false,
+  metro: false
+};
+
+const generateFetchURL = function(publicationList) {
+  // default url array
+  const defaultArray = ["daily-mail", "mirror", "metro"];
+
+  // create an array from object using key values
+  let publicationArray = Object.keys(publicationList);
+  // filter array to keys with 'true' values
+  let filteredArray = publicationArray.filter(function(pub) {
+    return publicationList[pub] === true;
+  });
+
+  // compile fetch url
+  let defaultArrayUrl = `&sources=${defaultArray}`;
+  let filteredPublicationUrl = `&sources=${filteredArray}`;
+  let fullURL = "";
+
+  // check if filteredArray has values
+  if (filteredArray.length == 0) {
+    fullURL = `${baseUrl}${defaultArrayUrl}`;
+  } else {
+    fullURL = `${baseUrl}${filteredPublicationUrl}`;
+  }
+  // CHANGE filteredArray GOOD
+  // console.log(filteredArray);
+
+  goFetch(fullURL);
+};
+
+/*
+----------------------
+CREATE CHECKBOX FILTER
+----------------------
+*/
+const createCheckboxFilter = function() {
+  checkboxArray.forEach(function(input) {
+    input.addEventListener("change", function(event) {
+      // new assigned value to match object key
+      // assign object value if checked is true
+      if (event.target.checked === true) {
```

Comment:
This could be shortened to `publicationList[event.target.value] = event.target.checked`